src/main/java/org/swisspush/redisques/RedisQues.java

```diff
@@ -451,54 +464,62 @@ private void readQueue(final String queue) {
+                                        if (myQueues.isEmpty()) {
+                                            stoppedHandler.handle(null);
+                                        }
+                                    }
+                                    // Issue notification to consume next message if any
+                                    String key2 = queuesPrefix + queue;
+                                    if (log.isTraceEnabled()) {
+                                        log.trace("RedisQues read queue: " + key);
+                                    }
+                                    redisClient.llen(key2, answer1 -> {
+                                        if (answer1.result() > 0) {
+                                            notifyConsumer(queue);
+                                        }
+                                    });
+                                });
+                            } else {
+                                // Failed. Message will be kept in queue and retried later
+                                log.debug("RedisQues Processing failed for queue " + queue);
+                                myQueues.put(queue, QueueState.READY);
+                                vertx.cancelTimer(sendResult.timeoutId);
+                                rescheduleSendMessageAfterFailure(queue);
                             }
-                            // Issue notification to consume next message if any
-                            String key2 = queuesPrefix + queue;
-                            if (log.isTraceEnabled()) {
-                                log.trace("RedisQues read queue: " + key);
-                            }
-                            redisClient.llen(key2, answer1 -> {
-                                if (answer1.result() > 0) {
-                                    notifyConsumer(queue);
-                                }
-                            });
                         });
                     } else {
-                        // Failed. Message will be kept in queue and retried later
-                        log.debug("RedisQues Processing failed for queue " + queue);
+                        // This can happen when requests to consume happen at the same moment the queue is emptied.
+                        log.debug("Got a request to consume from empty queue " + queue);
                         myQueues.put(queue, QueueState.READY);
-                        vertx.cancelTimer(sendResult.timeoutId);
-                        rescheduleSendMessageAfterFailure(queue);
                     }
                 });
             } else {
-                // This can happen when requests to consume happen at the same moment the queue is emptied.
-                log.debug("Got a request to consume from empty queue " + queue);
+                log.debug("Got a request to consume from locked queue " + queue);
                 myQueues.put(queue, QueueState.READY);
```

Comment:
Is it correct to change the queue state here? I think that the locked state should not modify this.